models/MigrationService.cfc

```diff
@@ -1,23 +1,74 @@
 component singleton accessors="true" {
 
-	property name="wirebox" inject="wirebox";
+	property name="wirebox" 			inject="wirebox";
+	property name="environment" 		inject="coldbox:setting:environment" default="development";
+	property name="manager"             default="cfmigrations.models.QBMigrationManager";
 	property name="migrationsDirectory" default="/resources/database/migrations";
-	property name="datasource";
-	property name="defaultGrammar"  default="AutoDiscover@qb";
-	property name="schema"          default="";
-	property name="migrationsTable" default="cfmigrations";
+	property name="seedsDirectory"      default="/resources/database/seeds";
+	property name="seedEnvironments"    default="development";
+	property name="managerProperties";
+
+
+	MigrationService function init() {
+		variables.managerProperties = {};
+		var args = arguments;
+		args.keyArray().each( function( key ){
+			if( isSimpleValue( args[ key ] ) ){
+				variables[ key ] = args[ key ];
+			} else if( key == "properties" ){
+				variables.managerProperties = args[ key ];
+			}
+		});
+
+		if( isSimpleValue( variables.seedEnvironments) ) variables.seedEnvironments = listToArray( variables.seedEnvironments );
+
+		return this;
+	}
+
+	function onDIComplete() {
+		variables.manager = variables.wirebox.getInstance( name=variables.manager, initArguments=variables.managerProperties );
+	}
+
+	public void function install( runAll = false ) {
```

Comment:
You guys really need to start documenting your functions. Nobody can be in everybody's mind.  So giving context to the code via the documentation of the function goes a long way. From input arguments, to returns to purpose.

Reply:
Fair enough.  FWIW, I added a ton of additional documentation to most of the methods in the Manager. :)

Reply:
Done